.github/workflows/build-witness-test.yml

```diff
@@ -0,0 +1,29 @@
+name: Build Witness CLI Test
+
+on:
+  pull_request:
+    branches: [ main ]
+  push:
+    branches: [ main ]
+
+jobs:
+  build-witness:
+    name: Build Witness CLI with local code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "stable"
```

Comment:
We are using a pinned version on witness build. That might be better here too: https://github.com/in-toto/go-witness/blob/762a5c5f431b8a5fba326c6bd3236c82c70a70ff/.github/workflows/witness.yml#L59